platform/src/main/java/org/hillview/storage/FileSetDescription.java

```diff
@@ -129,9 +129,11 @@ public ITable load() {
                     loader = new HillviewLogs.LogFileLoader(this.pathname);
                     break;
                 case "genericlog":
-                    GenericLogs genLog = new GenericLogs(FileSetDescription.this.getLogFormat());
-                    loader = new GenericLogs.LogFileLoader(this.pathname, genLog);
-                    break; 
+                    String format = FileSetDescription.this.getLogFormat();
+                    assert format != null;
```

Comment:
@mbudiu-vmw I think the best way to handle this is to make the "logFormat" field mandatory in the UI as if the user does not provide anything in the logFormat it would halt while loading.
![assert](https://user-images.githubusercontent.com/22631594/46864864-2b513480-ce39-11e8-9243-6f6bee33fc67.PNG)

Reply:
We cannot easily validate the logformat in the UI.
Here we just assert it is not null - that makes the style checker from IntelliJ happy. 
Otherwise, because of the way we use the `@Nullable` annotation, we would have to initialize the corresponding field of the log format with some value, even if that value is never used (the format is populated by reflection when loading form JSON).

Reply:
We can make the required fields mandatory so that there will not any null values for the fields.
https://www.w3schools.com/tags/att_input_required.asp

Reply:
We can fix this as part of https://github.com/vmware/hillview/issues/456.